src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -162,7 +163,7 @@ private MarcWithHoldingsRequestHelper() {
       }
 
       updateRequestMetadataFuture.onSuccess(res -> {
-        boolean isFirstBatch = resumptionToken == null || request.getRequestId() == null;
```

Comment:
The second condition is unnecessary here. The first one tells the same. Request object can contain both at once or no one of them.